charts/risingwave/templates/autoscaler.yaml

```
@@ -0,0 +1,35 @@
+{{/*
+  Copyright RisingWave Labs.
+  SPDX-License-Identifier: APACHE-2.0
+  */}}
+
+{{- if .Values.computeComponent.autoscaling.enabled -}}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "risingwave.computeComponentName" . }}-autoscaler
+  namespace: {{ .Release.Namespace }}
+  {{- if .Values.commonLabels }}
+  labels:
+    {{- tpl (toYaml .Values.commonLabels) . | nindent 4 }}
+  {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations:
+    {{- tpl (toYaml .Values.commonAnnotations) . | nindent 4 }}
+  {{- end }}
```

Forgot to mention, there are two helper functions to add the labels and annotations. Prefer using them instead.

```suggestion
  labels:
    {{- include "risingwave.labels" . | nindent 4 }}
    risingwave.risingwavelabs.com/component: compute
  {{- $annotations := (include "risingwave.annotations" . ) | trim }}
  {{- if $annotations }}
  annotations:
    {{ nindent 4 $annotations }}
  {{- end }}
```